packages/core/src/application/application-factory.ts

```diff
@@ -88,7 +88,7 @@ class AppFactory {
             container,
             appTypeOrMiddlewares as Array<express.RequestHandler>,
           );
-          return app as AppExpress;
```

Comment:
I can't approve this request to remove the `return app as IApplication` within each switch block, The reason why is because we will have multiple interfaces, as commented in the code you can see that we can possibly have a fastify interface, and we will have much more in the future.

The return has to be constrained within each switch case.